### PR TITLE
Frontend/nav menu improvement

### LINKED ISSUE #202: Improve Navigation Menu -> make the whole button area clickable
Currently links are only clickable if the user clicks exactly on the text as highlighted on the area:
image

Any slight deviation doesnt make the Link click, make the whole area clickable

Acceptance Criteria:
Make the whole area of a navigation button clickable:
image

### PR DESCRIPTION
**Pull request by ds161833**
Now the whole Link area is clickable not only the link's text:
image

### CODE REVIEW

File: Client/src/Components/ListRouter.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+export const linkWidth: number = 240
+
+export const ListItemLink = (props: IProps) => {
+  const { icon, primary, to } = props
+
+  const renderLink = React.useMemo(
+    () =>
+      // eslint-disable-next-line react/display-name
```
Comment (JeremiahTee) [inline Client/src/Components/ListRouter.tsx:41]: Is there a reason for having this commented? Testing purposes? Can we delete this line if it's useless?

File: Client/src/Components/NavigationMenu.tsx [hunk 2]
```diff
@@ -1,46 +1,20 @@
...
-import { AppBar, Box, Button, Divider, Drawer, Grid, IconButton, List, ListItem, ListItemIcon, Toolbar, Typography, makeStyles } from "@material-ui/core"
-import {
-  HashRouter,
-  Link,
-  NavLink,
-  Route
-} from "react-router-dom"
+import { AppBar, Box, Button, Divider, Drawer, Grid, IconButton, Toolbar, Typography, makeStyles } from "@material-ui/core"
+import { HashRouter, Link } from 'react-router-dom'
+import { ListRouter, getRoutedViews } from "./ListRouter"
```
Comment (JeremiahTee) [inline Client/src/Components/NavigationMenu.tsx:5]: Good code quality here. Everything is listed in one line as opposed to the previous lines.

File: Client/src/Components/NavigationMenu.tsx [hunk 3]
```diff
@@ -1,46 +1,20 @@
...
 import clsx from "clsx"
+import { linkWidth } from './ListRouter'
+import { signInRoute } from "../Consts/Routes"
 import universitylogo from './universitylogo.png'
-import PersonIcon from '@material-ui/icons/Person';
-import SignIn from "../Authentication/SignIn"
-import SignUp from "../Authentication/SignUp"
 
-const drawerWidth = 240
+const drawerWidth = linkWidth
```
Comment (JeremiahTee) [inline Client/src/Components/NavigationMenu.tsx:15]: Ok, so this is the main change. Also good job cleaning up the rest of the boilerplate code. This is what increases our good quality

**Review by JeremiahTee — CHANGES_REQUESTED** [review]
I tested the code on Windows 10, without docker. There is no issue. I also looked over the code on VSCode to make sure that there are no unused import statements. In a previous sprint from Release 1, I found out through Sonarqube that there were **many unused imports in the front-end**. In order for this technical debt not to accumulate, **PR reviewers** should **make sure that they check for unused imports** during review - so that they don't get merged (and subsequently flagged by Sonarqube as code smells). 

In terms of code quality, I like how some code has been refactored, for example line 73 -> the method "getRoutedViews()" is used. This makes it easier for developers to read code and understand what's going on when "repetitive code" is encapsulated by a method.

My only real comment is about the commented line at line 41, otherwise good job!

File: Client/src/Components/ListRouter.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import React from 'react'
+import { ResearchPaperAnalysisView } from "./ResearchPaperAnalysis/ResearchPaperAnalysisView"
+import { Route } from 'react-router'
+import SearchIcon from '@material-ui/icons/Search'
+import SearchView from "./Search/SearchView"
+import SignIn from "../Authentication/SignIn"
+import SignUp from "../Authentication/SignUp"
+
+interface IProps {
+  icon?: React.ReactElement;
```
Comment (GD-Dheer) [inline Client/src/Components/ListRouter.tsx:29]: The icon is optional?

**Review by GD-Dheer — APPROVED** [review]
A-OK

File: Client/src/Components/ListRouter.tsx (hunk 1 above)
Comment (ds161833) [reply Client/src/Components/ListRouter.tsx:41]: yeah you right, this comment disables eslint rule just for the next line, however I can disable this rule for the whole project, ill remove it

**Comment by ds161833** [comment]
> I tested the code on Windows 10, without docker. There is no issue. I also looked over the code on VSCode to make sure that there are no unused import statements. In a previous sprint from Release 1, I found out through Sonarqube that there were **many unused imports in the front-end**. In order for this technical debt not to accumulate, **PR reviewers** should **make sure that they check for unused imports** during review - so that they don't get merged (and subsequently flagged by Sonarqube as code smells).
> 
> In terms of code quality, I like how some code has been refactored, for example line 73 -> the method "getRoutedViews()" is used. This makes it easier for developers to read code and understand what's going on when "repetitive code" is encapsulated by a method.
> 
> My only real comment is about the commented line at line 41, otherwise good job!

I set up the environment in such a way that it removes unused imports for the file you're working on on save :) If you're still getting some that means not everyone has the eslint and 2 other extensions for vscode I added

File: Client/src/Components/ListRouter.tsx (hunk 1 above)
Comment (ds161833) [reply Client/src/Components/ListRouter.tsx:41]: it was removed, however not showing up in the conversation section of the PR review, take a look at the Files changed -> my last commit

Comment (JeremiahTee) [reply Client/src/Components/ListRouter.tsx:41]: Great thanks!

**Comment by JeremiahTee** [comment]
> > I tested the code on Windows 10, without docker. There is no issue. I also looked over the code on VSCode to make sure that there are no unused import statements. In a previous sprint from Release 1, I found out through Sonarqube that there were **many unused imports in the front-end**. In order for this technical debt not to accumulate, **PR reviewers** should **make sure that they check for unused imports** during review - so that they don't get merged (and subsequently flagged by Sonarqube as code smells).
> > In terms of code quality, I like how some code has been refactored, for example line 73 -> the method "getRoutedViews()" is used. This makes it easier for developers to read code and understand what's going on when "repetitive code" is encapsulated by a method.
> > My only real comment is about the commented line at line 41, otherwise good job!
> 
> I set up the environment in such a way that it removes unused imports for the file you're working on on save :) If you're still getting some that means not everyone has the eslint and 2 other extensions for vscode I added

You're right, I remember you guided me and helped me set up those extensions. I'll make sure to mention those extensions in my next Sonarqube report on Confluence. I will also make sure to ping teammates who have committed or left unused imports in case they show up on Sonarqube

**Review by JeremiahTee — APPROVED** [review]

**Pull request merged**